internal/metrics/metrics.go

```diff
@@ -5,13 +5,117 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 const (
 	MetricsEndpoint = "0.0.0.0:9090"
 )
 
+var (
+	EventsCounter *prometheus.CounterVec
+
+	ConditionCounter        *prometheus.CounterVec
+	ConditionRunTimeSummary *prometheus.SummaryVec
+
+	ActionCounter        *prometheus.CounterVec
+	ActionRuntimeSummary *prometheus.SummaryVec
+
+	ActionHandlerCounter        *prometheus.CounterVec
+	ActionHandlerRunTimeSummary *prometheus.SummaryVec
+
+	DownloadBytes          *prometheus.CounterVec
+	DownloadRunTimeSummary *prometheus.SummaryVec
+	UploadBytes            *prometheus.CounterVec
+	UploadRunTimeSummary   *prometheus.SummaryVec
+
+	StoreQueryErrorCount *prometheus.CounterVec
+)
+
+func init() {
+	EventsCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "flasher_events_received",
+			Help: "A counter metric to measure the total count of events received",
+		},
+		[]string{"valid", "response"}, // valid is true/false, response is ack/nack
+	)
+
+	ConditionCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "flasher_conditions_received",
+			Help: "A counter metric to measure the total count of condition requests received, successful and failed",
+		},
+		[]string{"condition", "state"},
+	)
+
+	ConditionRunTimeSummary = promauto.NewSummaryVec(
```

Comment:
Does this duplicate the `ActionRuntimeSummary`?

Reply:
Not really, a `firmwareInstall` `Condition` may turn into one or more `Action`s  where each applies firmware for a single hardware component.